### PR TITLE
Variant collection filtering

### PR DESCRIPTION
**Pull request by iskandr**
- split file loading from constructing `VariantCollection` objects (use `load_variants` function instead)
- added `clone` method to `VariantCollection` to simplify adding other filter methods 
- moved `drop_duplicates` from a file loading argument to be a method on `VariantCollection`
- added test for `drop_duplicates`

### CODE REVIEW

File: varcode/maf.py [hunk 1]
```diff
@@ -67,4 +68,44 @@ def load_maf_dataframe(filename, nrows=None, verbose=False):
...
+    elif len(ncbi_builds) > 1:
+        raise ValueError(
+            "Multiple NCBI builds (%s) for MAF file %s" % (ncbi_builds, filename))
+
+    reference_name = ncbi_builds[0]
+    variants = []
+
+    for _, x in maf_df.iterrows():
+        start_pos = x.Start_Position
+        end_pos = x.End_Position
```
Comment (tavinathanson) [inline varcode/maf.py]: Unused?

**Comment by tavinathanson** [comment]
What `pyensembl` dependency changed?

File: varcode/variant_collection.py [hunk 2]
```diff
@@ -1,122 +1,129 @@
...
+            original_filename=self.original_filename,
+            reference_path=self.reference_path,
+            reference_name=self.reference_name,
+            ensembl_release=self.ensembl_release)
+
+    def drop_duplicates(self):
+        """
+        Create a new VariantCollection without any duplicate variants.
+        """
+        return self.clone(set(self.variants))
```
Comment (tavinathanson) [inline varcode/variant_collection.py]: No `__hash__` for `Variant`?

File: test/test_drop_duplicates.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from varcode import Variant, VariantCollection
+
+def test_drop_duplicates():
+    v1 = Variant("1", 3000, "A", "G")
```
Comment (tavinathanson) [inline test/test_drop_duplicates.py]: Test for the case where `v1 = Variant("1", 3000, "A", "G")`, `v2 = Variant("1", 3000, "A", "G")` (but different objects)?

**Comment by tavinathanson** [comment]
Looks good to me aside from comments

File: varcode/variant_collection.py (hunk 2 above)
Comment (iskandr) [inline varcode/variant_collection.py]: Oh, I thought there already was one! I guess I forgot to add it when I adding `__hash__` and `__eq__` to all the PyEnsembl objects.

File: varcode/maf.py (hunk 1 above)
Comment (iskandr) [inline varcode/maf.py]: Added an assert that the expected length matches the given one in the MAF file.

File: test/test_drop_duplicates.py (hunk 3 above)
Comment (iskandr) [inline test/test_drop_duplicates.py]: :+1:

**Pull request merged**